modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -124,7 +160,21 @@ private String _formatSuper(String javaMethodContent) {
 		return StringUtil.replace(javaMethodContent, methodCall, newMethodCall);
 	}
 
-	private static final Pattern _superPattern = Pattern.compile(
-		"super.\\s*\\w+\\(\\s*.+\\)");
+	private String _getExtendedClassType(String content) throws Exception {
+		Matcher matcher = _extendedClassTypePattern.matcher(content);
+
+		if (!matcher.find()) {
+			throw new Exception("Cannot find extended class type");
+		}
+
+		return matcher.group(1);
+	}
+
+	private static final Pattern _extendedClassTypePattern = Pattern.compile(
+		"extends\\s*\\w+\\<(.+?)\\>");
+	private static final Pattern _superMethodPattern = Pattern.compile(
+		"(\\n?\\s*)super\\.\\s*(onAfterUpdate|onBeforeUpdate)\\(\\s*\\w+\\)");
```

Comment:
(\\n?\\s*) <---- we need to remember the spacing amount so the created clone can be on the same tab line.
super\\.\\s* <--- just make sure we find the super call and account for potential spacing
(onAfterUpdate|onBeforeUpdate) <---- only upgrade if one of these two calls are being made
\\(\\s* <--- function call and account for more whitespacing
\\w+ <---- this will guarantee we only match if exactly one parameter is inputed, not 0 or 2 or more
\\) <--- closing function call paren